irc_cap.c

```diff
@@ -0,0 +1,174 @@
+/********************************************************************\
+  * BitlBee -- An IRC to other IM-networks gateway                     *
+  *                                                                    *
+  * Copyright 2002-2013 Wilmer van der Gaast and others                *
+  \********************************************************************/
+
+/* IRCv3 CAP command                                                    */
+
+/*
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License with
+  the Debian GNU/Linux distribution in /usr/share/common-licenses/GPL;
+  if not, write to the Free Software Foundation, Inc., 51 Franklin St.,
+  Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "bitlbee.h"
+
+typedef struct {
+	char *name;
+	irc_cap_flag_t flag;
+} cap_info_t;
+
+static const cap_info_t supported_caps[] = {
+	{"foo", CAP_FOO},
+	{"bar", CAP_BAR},
+	{NULL},
+};
+
+static irc_cap_flag_t cap_flag_from_string(char *cap_name)
+{
+	int i;
+
+	if (!cap_name && !cap_name[0]) {
+		return 0;
+	}
+
+	if (cap_name[0] == '-') {
+		cap_name++;
+	}
+
+	for (i = 0; supported_caps[i].name; i++) {
+		if (strcmp(supported_caps[i].name, cap_name) == 0) {
+			return supported_caps[i].flag;
+		}
+	}
+	return 0;
+}
+
+static gboolean irc_cmd_cap_req(irc_t *irc, char *caps)
+{
+	int i;
+	char *lower = NULL;
+        char **split = NULL;
```

Comment:
Whoa, I didn't know I was capable of making this kind of mistake (it should be pretty damn obvious and hard to make with my editor configs)